connectors/katalog/pkg/connector/connector.go

```diff
@@ -177,3 +196,63 @@ func (r *Handler) deleteAsset(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+// Enables deletion of assets to katalog.
+func (r *Handler) updateAsset(c *gin.Context) {
+	// Parse request
+	var request datacatalog.UpdateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		errString := "Error during ShouldBindJSON in updateAsset"
+		r.log.Info().Msg(errString + err.Error())
+		r.reportError(c, http.StatusBadRequest, errString)
+		return
+	}
+	logging.LogStructure("UpdateAssetRequest received:", request, &r.log, zerolog.DebugLevel, false, false)
+
+	splittedID := strings.SplitN(string(request.AssetID), "/", 2)
+	if len(splittedID) != 2 {
+		errorMessage := fmt.Sprintf("UpdateAssetRequest has an invalid asset ID %s (must be in namespace/name format)", request.AssetID)
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMessage})
+		return
+	}
+	namespace, name := splittedID[0], splittedID[1]
+
+	asset := &v1alpha1.Asset{}
+	if err := r.client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, asset); err != nil {
+		if errors.IsNotFound(err) {
+			errString := "Error: Asset Not Found during updateAsset"
+			r.log.Info().Msg(errString + err.Error())
+			c.JSON(http.StatusNotFound, gin.H{"error": errString})
+			return
+		}
+		errString := "Error during getting asset information during update asset"
+		r.log.Info().Msg(errString + err.Error())
```

Comment:
just err.Error() is enough, the prefix is too long and not needed

Reply:
Updated. Thanks.